src/replication.h

```diff
@@ -28,28 +30,101 @@ enum ReplState {
 };
 
 typedef std::function<void(const std::string, const uint32_t)> fetch_file_callback;
+class FeedSlaveThread;
 
-class FeedSlaveThread {
+class TransmitStartEvent : public ObserverEvent {
+ public:
+  TransmitStartEvent() : ObserverEvent() {}
+  int conn_fd = 0;
+  FeedSlaveThread* thread_ptr = nullptr;
+};
+
+class BeforeSendEvent : public ObserverEvent {};
+
+class AfterSendEvent : public ObserverEvent {
+ public:
+  AfterSendEvent() : ObserverEvent() {}
+  int conn_fd = 0;
+  uint64_t sequenceNumber = 0;
+};
+
+class TransmitEndEvent : public ObserverEvent {
+ public:
+  TransmitEndEvent() : ObserverEvent() {}
+  FeedSlaveThread* thread_ptr = nullptr;
+};
+
+class SyncStatusChangeEvent : public ObserverEvent {
+ public:
+  SyncStatusChangeEvent() : ObserverEvent() {}
+  int conn_fd = 0;
+  uint64_t lastSequenceNumberEnd = 0;
+};
+
+class FeedSlaveHandler : public EventHandler {
+ public:
+  FeedSlaveHandler() : EventHandler() {
+    registerEventHandler<TransmitStartEvent>(
+      std::bind(&FeedSlaveHandler::transmit_start, this, std::placeholders::_1, std::placeholders::_2));
+    registerEventHandler<BeforeSendEvent>(
+      std::bind(&FeedSlaveHandler::before_send, this, std::placeholders::_1, std::placeholders::_2));
+    registerEventHandler<AfterSendEvent>(
+      std::bind(&FeedSlaveHandler::after_send, this, std::placeholders::_1, std::placeholders::_2));
+    registerEventHandler<TransmitEndEvent>(
+      std::bind(&FeedSlaveHandler::transmit_end, this, std::placeholders::_1, std::placeholders::_2));
+  }
+
+ private:
+  void transmit_start(Observable subject, ObserverEvent const& event);
```

Comment:
Can you help to change this function into the camel case?  since we use the camel case to name the function and use the snake case in the variable.